grid_map_demos/src/OctomapToGridmapDemo.cpp

```diff
@@ -37,31 +44,51 @@ OctomapToGridmapDemo::~OctomapToGridmapDemo()
 
 bool OctomapToGridmapDemo::readParameters()
 {
-  nodeHandle_.param("octomap_service_topic", octomapServiceTopic_, std::string("/octomap_binary"));
-  nodeHandle_.param("min_x", minX_, NAN);
-  nodeHandle_.param("max_x", maxX_, NAN);
-  nodeHandle_.param("min_y", minY_, NAN);
-  nodeHandle_.param("max_y", maxY_, NAN);
-  nodeHandle_.param("min_z", minZ_, NAN);
-  nodeHandle_.param("max_z", maxZ_, NAN);
+  this->declare_parameter("octomap_service_topic", std::string("/octomap_binary"));
+  this->declare_parameter("min_x", NAN);
+  this->declare_parameter("max_x", NAN);
+  this->declare_parameter("min_y", NAN);
+  this->declare_parameter("max_y", NAN);
+  this->declare_parameter("min_z", NAN);
+  this->declare_parameter("max_z", NAN);
+
+  this->get_parameter("octomap_service_topic", octomapServiceTopic_);
+  this->get_parameter("min_x", minX_);
+  this->get_parameter("max_x", maxX_);
+  this->get_parameter("min_y", minY_);
+  this->get_parameter("max_y", maxY_);
+  this->get_parameter("min_z", minZ_);
+  this->get_parameter("max_z", maxZ_);
   return true;
 }
 
 void OctomapToGridmapDemo::convertAndPublishMap()
 {
-  octomap_msgs::GetOctomap srv;
-  if (!client_.call(srv)) {
-    ROS_ERROR_STREAM("Failed to call service: " << octomapServiceTopic_);
+  while (!client_->wait_for_service(std::chrono::seconds(1))) {
+    if (!rclcpp::ok()) {
+      RCLCPP_ERROR(this->get_logger(), "client interrupted while waiting for service to appear.");
+      return;
+    }
+    RCLCPP_INFO(this->get_logger(), "waiting for service to appear...");
```

Comment:
Wouldn't that get spammed? Is there a throttle version?

Reply:
Fixed.